app/Models/User.php

```diff
@@ -2043,6 +2043,14 @@ public function profileBeatmapsetsLoved()
             ->with('beatmaps');
     }
 
+    public function profileBeatmapsetsGuest()
+    {
+        return Beatmapset::whereNotIn('beatmapset_id', $this->beatmapsets->pluck('beatmapset_id'))
+            ->whereHas('beatmaps', fn ($q) => $q->where('user_id', $this->getKey()))
+            ->active()
+            ->with('beatmaps');
```

Comment:
as mentioned in the comment, this should hide pending and lower state maps. Also I haven't tested it but this probably should be `where('user_id', '<>', $userId)` instead of `whereNotIn(...)` (or at least use subquery instead of separate query).